scripts/utils/trading_strategy_helpers.js

```diff
@@ -244,34 +242,30 @@ module.exports = function(web3 = web3, artifacts = artifacts) {
     log(`Lowest-Highest Limit ${lowestLimit}-${highestLimit}`)
 
     const stepSizeAsMultiplier = Math.pow(highestLimit / lowestLimit, 1 / bracketAddresses.length)
-    const transactions = []
     log(
       `Constructing bracket trading strategy order data based on valuation ${targetPrice} ${stableToken.symbol} per ${targetToken.symbol}`
     )
-    let currentLowerPriceLimit = lowestLimit
 
-    for (let bracketIndex = 0; bracketIndex < bracketAddresses.length; bracketIndex++) {
-      const bracketAddress = bracketAddresses[bracketIndex]
+    const transactions = await Promise.all(bracketAddresses.map(async (bracketAddress, bracketIndex) => {
       assert(await isOnlySafeOwner(masterAddress, bracketAddress), "each bracket should be owned only by the master Safe")
 
-      const lowerLimit = currentLowerPriceLimit
-      const upperLimit = currentLowerPriceLimit * stepSizeAsMultiplier
+      const lowerLimit = lowestLimit * Math.pow(stepSizeAsMultiplier, bracketIndex)
+      const upperLimit = lowerLimit * stepSizeAsMultiplier
 
       const [upperSellAmount, upperBuyAmount] = calculateBuyAndSellAmountsFromPrice(upperLimit, targetToken)
       // While the first bracket-order trades standard_token against target_token, the second bracket-order trades
       // target_token against standard_token. Hence the buyAmounts and sellAmounts are switched in the next line.
       const [lowerBuyAmount, lowerSellAmount] = calculateBuyAndSellAmountsFromPrice(lowerLimit, targetToken)
 
-      log(`Safe ${bracketIndex} - ${bracketAddress}:\n  Buy  ${targetToken.symbol} with ${stableToken.symbol} at ${lowerLimit}`)
-      log(`  Sell ${targetToken.symbol} for  ${stableToken.symbol} at ${upperLimit}`)
+      log(`Safe ${bracketIndex} - ${bracketAddress}:\n  Buy  ${targetToken.symbol} with ${stableToken.symbol} at ${lowerLimit}\n  Sell ${targetToken.symbol} for  ${stableToken.symbol} at ${upperLimit}`)
       const buyTokens = [targetTokenId, stableTokenId]
       const sellTokens = [stableTokenId, targetTokenId]
-      const validFroms = [batch_index + validFrom, batch_index + validFrom]
+      const validFroms = [await batchIndexPromise + validFrom, await batchIndexPromise + validFrom]
```

Comment:
It would be good to await this promise just once for performance reasons. Actually, I would prefer to await the batchIndex only once for all orders, so that they all have the same validity.

Reply:
I believe that awaiting an already resolved promise is very efficient, and an improvement would most likely be non noticeable by the user (my guess is <1ms, but I didn't test it).
Moreover, the output of resolving the same promise should always be the same for all orders, independently of when each of the promises are awaited, so is should never happen that the batchIndices are different.